src/main/java/com/openlattice/authorization/HazelcastAuthorizationService.java

```diff
@@ -112,11 +111,30 @@ public void addPermission(
                 new PermissionMerger( permissions, securableObjectType, expirationDate ) );
     }
 
+    private void ensureAclKeysHaveOtherUserOwners( Set<AclKey> aclKeys, Set<Principal> principals ) {
```

Comment:
This is a lot of overhead as it requires locally loading a bunch of data on every request. 

A more efficient way of implementing this is to use a fast aggregator (Aggregators.count()) with the predicate p below and another predicate that only checks for owner and aclKeys. If the counts are different than it is safe to delete and requires way less serialization and over head.

Reply:
updated